x-pack/plugins/endpoint/common/generate_data.ts

```diff
@@ -90,8 +90,17 @@ export class EndpointDocGenerator {
     this.policy = this.randomChoice(POLICIES);
   }
 
-  public randomizeIPs() {
+  // This function will create new values for all the host fields, so documents from a different endpoint can be created
+  // This provides a convenient way to make documents from multiple endpoints that are all tied to a single seed value
+  public randomizeHostData() {
```

Comment:
We could probably have the constructor call this function so the calls aren't duplicated.

Reply:
Good idea.  The compiler complained that the class variables weren't definitely assigned when I called it as it was so I pulled the fields into a new structure.  It's now returned by the private function below and assigned in this function and the constructor.